sympy/sets/tests/test_fancysets.py

```diff
@@ -77,8 +77,10 @@ def test_integers():
 def test_ImageSet():
     raises(ValueError, lambda: ImageSet(x, S.Integers))
     assert ImageSet(Lambda(x, 1), S.Integers) == FiniteSet(1)
-    assert ImageSet(Lambda(x, y), S.Integers
-        ) == {y}
+    assert ImageSet(Lambda(x, y), S.Integers) == {y}
+    assert ImageSet(Lambda(x, 1), S.EmptySet) == S.EmptySet
+    empty = Intersection(FiniteSet(log(2)/pi), S.Integers)
+    assert ImageSet(Lambda(x, 1), empty) != FiniteSet(1)  # issue #17471
```

Comment:
What is the result here? Does it not evaluate? If so I think it would be better to use unchanged i.e.:
```python
assert unchanged(ImageSet, Lambda(x, 1), empty)
```

Reply:
Ok! I didn't want to let the test fail if `empty` is properly recognized as being empty in the future, but it's probably better to have the test be more specific.